src/test/java/com/tahera/service/BookServiceTest.java

```diff
@@ -79,4 +76,27 @@ public void shouldNotHaveUpdateBookWhenItDoesnotExists() {
 
         Mockito.verify(bookRepository, Mockito.never()).save(any());
     }
+
+    @Test
+    public void shouldSetBookAsRead() {
+        Book book = new Book(1L, "test", 1, false, "test");
+        Mockito.when(bookRepository.findByTitleAndAuthor(book.getTitle(), book.getAuthor())).thenReturn(Optional.of(book));
+        Book setBookAsRead = new Book();
+        setBookAsRead.setAuthor("test");
+        setBookAsRead.setTitle("test");
+        setBookAsRead.setRate(2);
+        setBookAsRead.setRead(true);
+
+        bookService.setBookRead(setBookAsRead);
```

Comment:
there is no assertion here, should it mockito verify save has been called?

Reply:
Great catch!!
Added: `Mockito.verify(bookRepository).save(bookCaptor.capture());`